Sources/Subs-Db-mysql.php

```diff
@@ -170,11 +170,10 @@ function smf_db_replacement__callback($matches)
 	if ($matches[1] === 'db_prefix')
 		return $db_prefix;
 
-	if (!empty($user_info))
+	if (!empty($user_info) && strpos($matches[1],'query_') !== false)
 	{
-		foreach (array_keys($user_info) as $key)
-			if (strpos($key, 'query_') !== false && $key === $matches[1])
-				return $user_info[$matches[1]];
+		if (isset($user_info[$matches[1]]))
```

Comment:
If we are checking `isset($user_info[$matches[1]])` we don't even need to check `!empty($user_info)`, so we could reduce this even further to just:
```
if (isset($user_info[$matches[1]]) && strpos($matches[1],'query_') !== false)
```